lib/iris/tests/test_analysis.py

```diff
@@ -351,102 +351,173 @@ def test_multi_coord_mdtol(self):
 
 
 class TestAggregators(tests.IrisTest):
-    def test_percentile_1d(self):
+
+    def _check_coord_properties(self, cube, collapse_coord, unit, points):
+        if isinstance(collapse_coord, six.string_types):
+            collapse_coord = [collapse_coord]
+        name = 'percentile_over_{}'.format('_'.join(collapse_coord))
+        self.assertTrue(cube.coord(name))
```

Comment:
Hm, we've lost the previous comments thread here, apparently... To recall some of that thread:

> A previous reviewer expressed a distaste for the CML checks

And with good reason, admittedly – they are too sensitive to small and unimportant changes; though they are good at checking a cube wholesale! Given that you've now introduced CML for the tests here can I talk you into dropping this check method? All the checks this makes are included in the CML...

For reference, I'm still not a fan of this particular assertion. I think that if the named coord you're looking for does not exist then `cube.coord(name)` will raise an exception, which will cause the test that called this method to error, which is undesirable. It might be better to do something like the following (if you don't replace this check method with CML checking):

```python
coord_names = [c.name() for c in cube.coords()]
self.assertIn(name, coord_names)
```

This won't error if the named coord doesn't exist, but that assertion will fail, which is a better way to be. It's also clearer to the test reader what the assertion is doing.

Reply:
You can definitely convince me of this, it was only there as the beginnings of some replacement to the CML checks, but with them back in place, it serves no useful purpose. I have removed the `_check_coord_properties` function along with the associated keyword. Now, if a `CML_filename` is provided it will conduct the CML test, otherwise it will just compare the expected values, as was previously the case.